src/eslint.ts

```diff
@@ -103,7 +103,18 @@ export class CachedESLint {
   }
 
   printResults(results: ESLint.LintResult[]): void {
-    const resultText = format(results);
+    // get used plugins from `results`
+    const plugins = scanUsedPluginsFromResults(results);
+
+    // get `rulesMeta` from `results`
+    const eslint = new ESLint({
+      ...this.defaultOptions,
+      overrideConfig: { plugins },
+    });
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const rulesMeta: ESLint.LintResultData['rulesMeta'] | undefined = (eslint as any).getRulesMetaForResults?.(results);
```

Comment:
- `getRulesMetaForResults` is a feature added in ESLint 7.29.0.
  - https://github.com/eslint/eslint/releases/tag/v7.29.0
- Therefore, the function may not exist in versions lower than 7.29.0.

Reply:
blocked: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/56950